### PR TITLE
[docs] First docs contribution.  Rename UCB -> LinUCB.

### PR DESCRIPTION
**Pull request by christy**
@maxpumperla 

## Related issue number
First docs contribution.  Part of #24347.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/rllib/rllib-algorithms.rst [hunk 1]
```diff
@@ -70,8 +70,8 @@ Algorithm                     Frameworks Discrete Actions        Continuous Acti
 .. _`+RNN`: rllib-models.html#rnns
 .. _`TD3`: rllib-algorithms.html#ddpg
 .. _`+Attention`: rllib-models.html#attention
-.. _`TS`: rllib-models.html#lints
-.. _`UCB`: rllib-models.html#linucb
+.. _`TS`: rllib-algorithms.html#lints
```
Comment (maxpumperla) [inline doc/source/rllib/rllib-algorithms.rst:73]: why did you change the `TS` link? Doesn't look right to me.

File: doc/source/rllib/rllib-algorithms.rst [hunk 2]
```diff
@@ -893,7 +899,7 @@ often: The "forward" model will eventually get better at predicting these next l
 
 
 
-.. _RE3:
+.. _re3:
```
Comment (maxpumperla) [inline doc/source/rllib/rllib-algorithms.rst:902]: what do we gain from this? did you check that still works?

File: rllib/README.rst [hunk 3]
```diff
@@ -53,6 +53,77 @@ Install RLlib and run your first experiment on your laptop in seconds:
     $ rllib train --run APPO --env CartPole-v0 --torch
 
 
+Algorithms Supported
+----------------------
+
+Offline RL:  
+
+- `Behavior Cloning (BC; derived from MARWIL implementation) <https://docs.ray.io/en/master/rllib/rllib-algorithms.html#bc>`__
```
Comment (maxpumperla) [inline rllib/README.rst:61]: I think it's much better in this scenario to use e.g.

```
:ref:`Behavior Cloning (BC; derived from MARWIL implementation) <bc>`
```

everywhere. The way you're doing it right now, users will always be redirected to `master`, even if they're on `latest`. Whenever possible, we want to have valid, relative cross-references.

**Review by maxpumperla — CHANGES_REQUESTED** [review]
overall looking great, just a few nits!

File: rllib/README.rst (hunk 3 above)
Comment (maxpumperla) [reply rllib/README.rst:61]: @christy can we change this, please?

File: doc/source/rllib/rllib-algorithms.rst (hunk 1 above)
Comment (christy) [reply doc/source/rllib/rllib-algorithms.rst:73]: `TS` link before was not pointing to the Thompson Sampling algorithm before.  Now it points there.

File: doc/source/rllib/rllib-algorithms.rst (hunk 2 above)
Comment (christy) [reply doc/source/rllib/rllib-algorithms.rst:902]: Checked it works.  Sphinx documentation said ref links are supposed to be lower case..

File: rllib/README.rst (hunk 3 above)
Comment (christy) [reply rllib/README.rst:61]: I made maybe 20 commits, trying to get :ref:`text <link>` Sphinx syntax to work in the README.rst file.  

I finally give up!  The existing syntax works.  Maybe later I can get the Sphinx :ref: syntax working.

**Review by maxpumperla — APPROVED** [review]

**Comment by maxpumperla** [comment]
@christy approved, but pending failing linter tests:

```
--
  | Warning, treated as error:
  | /ray/doc/source/rllib/rllib-offline.rst:70:Explicit markup ends without a blank line; unexpected unindent.
  | make: *** [Makefile:55: html] Error 2
```
